### ISSUE TITLE
Package installation requires cmake binary

### ISSUE THREAD

**Issue by TonyPapousekLC** [m-radzikowski/aws-sdk-client-mock#69]
**Checklist**

- [ x] I have read [Caveats](https://github.com/m-radzikowski/aws-sdk-client-mock#caveats) documentation and didn't find a solution for this problem there.

**Bug description**

A dependency requires aws-crt, which in turn requires cmake-js, which requires a local cmake installation. Which together makes this package harder to sell to the rest of the team.

Could you please list this requirement in the Install doc?

Relevant output from `npm install -D aws-sdk-client-mock -ddd` (`/path/to/project` replaces sensitive information):

```sh
npm ERR! path /path/to/project/node_modules/aws-crt
npm ERR! command failed
npm ERR! command sh -c node ./scripts/install.js
npm ERR! info TOOL Using Unix Makefiles generator.
npm ERR! info TOOL Building only the install target, as specified from the command line.
npm ERR! ERR! OMG CMake executable is not found. Please use your system's package manager to install it, or you can get installers from there: http://cmake.org.
npm ERR! /path/to/project/node_modules/cmake-js/lib/cMake.js:117
npm ERR!         throw new Error("CMake executable is not found. Please use your system's package manager to install it, or you can get installers from there: http://cmake.org.");
npm ERR!               ^
npm ERR!
npm ERR! Error: CMake executable is not found. Please use your system's package manager to install it, or you can get installers from there: http://cmake.org.
npm ERR!     at CMake.verifyIfAvailable (/path/to/project/node_modules/cmake-js/lib/cMake.js:117:15)
npm ERR!     at CMake.build (/path/to/project/node_modules/cmake-js/lib/cMake.js:287:10)
npm ERR!     at BuildSystem._invokeCMake (/path/to/project/node_modules/cmake-js/lib/buildSystem.js:70:40)
npm ERR!     at async buildFromRemoteSource (/path/to/project/node_modules/aws-crt/scripts/build.js:148:5)
```

**Environment**

- Node version: v16.13.0
- Testing lib and version: jest@27.4.3
- Typescript version: 4.4.3
- AWS SDK v3 Client mock version: 0.5.6, presumably
- AWS JS SDK libs and versions:
  - @aws-sdk/client-sqs: 3.43.0
- Platform: macOS 12.1, Apple M1 Pro chipset

**Comment by prince3339** [comment]
I'm also facing this issue. I recently switched to Macbook M1 Pro and now my app breaks. I used to be fine on my Intel MacBook Pro though.

**Comment by prince3339** [comment]
![image](https://user-images.githubusercontent.com/8009083/149764367-d6d24bc9-4203-4909-a0b3-6cbe50bf9f4a.png)

**Comment by m-radzikowski** [comment]
**Full explanation**

To enable mocking for `@aws-sdk/lib-storage` the library has a peer dependency on `@aws-sdk/client-s3`, which has an (indirect) peer dependency on `aws-crt`. The idea was that users that need `lib-storage` mocks have `client-s3` installed anyway, and others will not be affected.

However, NPM from version 7 [installs the peer dependencies automatically](https://github.blog/2021-02-02-npm-7-is-now-generally-available/). So NPM users will have `@aws-sdk/client-s3` and `aws-crt` installed. This is a pretty big issue with AWS SDK v3 itself too: https://github.com/aws/aws-sdk-js-v3/issues/2750, https://github.com/aws/aws-sdk-js-v3/issues/2822

Now, the best long-term solution is to remove the `@aws-sdk/client-s3` dependency from this library, as it also causes other problems thanks to the new NPM behavior (#56).

To still provide support for mocking `lib-storage` and similar, but not require all users to have `client-s3` installed, I need to move those mocks definitions out from the main package `index.ts`. I did implement it (#57), but to use it, package.json `exports` is required in test libraries like `jest`. It will be released in the next version, v28. Until then, I can't merge changes.

So right now, when using NPM, you will get `aws-crt` installed. And `aws-crt` requires `cmake` installed locally (#3137).

**TL;DR**

If you are using NPM to install packages, you need to have `cmake` installed locally. `brew install cmake` should do the trick on macOS, also on Macbooks M1.

In the future, after jest v28 release with package exports support, I will remove the dependencies on `@aws-sdk/*` libs, but it will require using jest v28 or higher, or package exports support in other testing frameworks.

For now, I will update the installation info to include `cmake` requirement in the README, as @TonyPapousekLC suggested, and close this issue. Feel free to write here if it does not solve the problem, and I will reopen.

**Comment by TonyPapousekLC** [comment]
I appreciate the detailed response, that makes sense.

**Comment by Antonio-Riccelli** [comment]
@m-radzikowski  To confirm, Jest is currently on version 29 - is cmake then no longer required? 
Many thanks.

**Comment by m-radzikowski** [comment]
@Antonio-Riccelli no, it's not since v1.0.0.

**Comment by VityaSchel** [comment]
I like how emotional and passive agressive this error is.

npm ERR! 😡 ERR! 🤬🤬🤬 OMG ❗️😦😨 CMake executable is not found. 😤
